src/usb_moded-dbus.c

```diff
@@ -159,7 +184,10 @@ static DBusHandlerResult msg_handler(DBusConnection *const connection, DBusMessa
 		else
 		{
 			/* error checking is done when setting configuration */
-			if(!set_mode_setting(config))
+			int ret = set_mode_setting(config);
+			if (ret == SET_CONFIG_UPDATED)
+				usb_moded_send_config_signal(MODE_SETTING_ENTRY, MODE_SETTING_KEY, config);
```

Comment:
In that case, the proposed `enum set_config_result_t` would degenerate into simply a `gboolean` if the only reason for having `SET_CONFIG_UNCHANGED` is to decide whether to send the signal.